### ISSUE TITLE
`pp.cumprod` does not normalize quaternion properly.

### ISSUE THREAD

**Issue by MarkChenYutian** [pypose/pypose#346]
## Summary

`pp.cumprod` method should add normalization to rotation (quaternion) to improve numerical stability and reduce loss of precision on long sequences.

## Explanation

Currently, `pp.cumprod` performs cumulative products on `LieTensor` directly. However, under long trajectories with violent rotation, such naive implementation tends to accumulate errors quickly. To reduce precision lost due to cumulative product, it is a common practice to normalize quaternion during computation.

Below I re-implement a draft version of `cumprod` for `SE3` that normalizes the quaternion and compares its result against `pp.cumprod`. As indicated in the plot, the current implementation can produce up to 10 times more errors than the proposed, normalized version of `cumprod`.

**Reference**

* https://gamedev.stackexchange.com/questions/110918/storing-component-vectors-to-avoid-matrix-and-quaternion-inaccuracies

**Minimal Reporducible Demo**

![image](https://github.com/pypose/pypose/assets/47029019/4b636d67-57b5-4a47-9d11-cfe83c611b7b)

Data: [poses.txt](https://github.com/user-attachments/files/15844304/poses.txt)

```python
import torch
import pypose as pp
import numpy as np
import matplotlib.pyplot as plt

def NormalizeQuat(x: pp.LieTensor) -> pp.LieTensor:
    data, ltype = x.tensor(), x.ltype
    data[..., 3:] = data[..., 3:]/data[..., 3:].norm(dim=-1, keepdim=True)
    return pp.LieTensor(data, ltype=ltype)

poses = torch.cat([pp.identity_SE3(1), pp.SE3(np.loadtxt("poses.txt"))], dim=0)
motions = poses[:-1].Inv() @ poses[1:]

cumprod_poses = pp.cumprod(motions, dim=0, left=False)
cumprod_diff = ((poses[1:].Log() - cumprod_poses.Log()).norm(dim=1))

normd_poses = pp.cumops(motions, dim=0, ops=lambda a, b: NormalizeQuat(a) @ NormalizeQuat(b))
normd_diff = ((poses[1:].Log() - normd_poses.Log()).norm(dim=1))


plt.plot(cumprod_diff, label="cumprod")
plt.plot(normd_diff, label="normed")
plt.ylabel("Error on se(3)")
plt.xlabel("#Frame")
plt.legend()
plt.show()

```

## Risks

May reduce the speed of `cumprod` slightly. But should not be significant since normalization will only be conducted for $O(log N)$ times on cumulative product of $N$ `LieTensor`s.

## Involved components

https://github.com/pypose/pypose/blob/931dc6f496936792059806eee190b9d6755c8c6f/pypose/basics/ops.py#L153
